### DISCUSSION TITLE
Custom validation on TitlePart

### DISCUSSION THREAD

**Discussion by djjohnsonplymouth** [OrchardCMS/OrchardCore#16816] [Q&A] [answered]
I have a ContentType which just has a TitlePart attached to it. I would like to ensure that the title on the TitlePart is unique when compared to other contentItems of the same ContentType. I have tried to override the ValidatingAsync methods on a custom ContentHandlerBase class where I can hook into the CreatedAsync override but ValidatingAsync doesn't get hit when creating the content item. I have also tried to create a custom ContentPartHandler<TitlePart> but none of the overrides (CreatedAsync, ValidatingAsync etc...) get executed. I have registered with DI with 
```
services.AddScoped<IContentPartHandler, CustomTitlePartHandler>();
services.AddScoped<IContentHandler, CustomReportContentHandler>();
```
Is this expected behaviour? 

Any help on how I might hook into the TitlePart validation to achieve the desired result would be great. Thank you.

**Comment by Piedone** [comment]
You need to register handlers with the `AddHandler` method.

Most possibly the `UpdatedAsync` handler method would be the suitable one for you.

**Comment by ns8482e** [comment, marked as answer]
For Admin UI, Validation happens in display drivers, use display driver for validating your custom part, make sure you add validation error to ModelState. You could reuse logic of `ValidatingAsync`  by calling from within your driver

**Reply by djjohnsonplymouth** [reply]
Thank you. It looks as though the handlers only execute under certain conditions. Using the display driver for validation was the right approach.
